File: integrations/aws-strands/python/examples/server/api/agentic_chat_multimodal.py
"""Agentic Chat with Multimodal support for AWS Strands.

Demonstrates multimodal message handling. When the user uploads an image,
the adapter converts AG-UI InputContent to Strands ContentBlock format
and passes it to the vision-capable model.
"""
import os
from pathlib import Path
from dotenv import load_dotenv

# Suppress OpenTelemetry context warnings
os.environ["OTEL_SDK_DISABLED"] = "true"
os.environ["OTEL_PYTHON_DISABLED_INSTRUMENTATIONS"] = "all"

from strands import Agent
from ag_ui_strands import StrandsAgent, create_strands_app
from server.model_factory import create_model
from server.settings import cors_origins

# Load environment variables from .env file
env_path = Path(__file__).parent.parent.parent / '.env'

load_dotenv(dotenv_path=env_path)

# Create model from MODEL_PROVIDER env var (default: openai)
model = create_model()

strands_agent = Agent(
    model=model,
    system_prompt="""
    You are a helpful assistant that can analyze images and documents.
    When the user shares an image, describe what you see in detail.
    When the user shares a document, summarize its contents.
    Always be descriptive and specific about visual content.
    """,
)

agui_agent = StrandsAgent(
    agent=strands_agent,
    name="agentic_chat_multimodal",
    description="Conversational Strands agent with multimodal content support",
)

app = create_strands_app(agui_agent, "/", origins=cors_origins())

File: integrations/aws-strands/typescript/examples/server/api/agentic-chat-multimodal.ts
/**
 * Agentic Chat with Multimodal support for AWS Strands (TypeScript).
 *
 * Demonstrates multimodal message handling. When the user uploads an image,
 * the adapter converts AG-UI InputContent to Strands ContentBlock format and
 * passes it to whichever model the factory built, which the operator is
 * responsible for pointing at something that can read images.
 */

import { Agent } from "@strands-agents/sdk";
import { StrandsAgent } from "@ag-ui/aws-strands";
import { createStrandsApp } from "@ag-ui/aws-strands/server";
import { createModel } from "../model-factory";
import { demoPort, listenOrExit, runIfMain } from "../run-if-main";

export const SYSTEM_PROMPT = `
    You are a helpful assistant that can analyze images and documents.
    When the user shares an image, describe what you see in detail.
    When the user shares a document, summarize its contents.
    Always be descriptive and specific about visual content.
  `;

export async function createAgenticChatMultimodalAgent(): Promise<StrandsAgent> {
  return new StrandsAgent({
    agent: new Agent({
      model: await createModel(),
      systemPrompt: SYSTEM_PROMPT,
    }),
    name: "agentic_chat_multimodal",
    description: "Conversational Strands agent with multimodal content support",
  });
}

runIfMain(import.meta.url, async () => {
  // Port first: it throws on a malformed PORT, and building the agent first
  // would surface a missing API key instead and hide the real complaint.
  const port = demoPort();
  const app = await createStrandsApp(await createAgenticChatMultimodalAgent(), {
    path: "/",
  });
  listenOrExit(app, "agentic-chat-multimodal", port);
});
